### PR TITLE
Added application certificate health check

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/django-server-status/issues/33

#### What's this PR do?
- adds certificate health check status

#### How should this be manually tested?
- on odl video add/replace this `HEALTH_CHECK = ['CELERY', 'REDIS', 'POSTGRES', 'CERTIFICATE']`in settings file, 
- Go to you local odl video status url
- Use token `STATUS_TOKEN`in url  from you settings
- Url will be like http://od.odl.local:8089/status/?token=statustoken

@pdpinch 
#### Screenshots (if appropriate)
screen shot 2018-09-13 at 7 05 21 pm

### CODE REVIEW

File: server_status/views.py [hunk 1]
```diff
@@ -152,6 +153,40 @@ def get_celery_info():
...
+    """
+    certificate_status = {
+        'app_cert_expires': '',
+        'status': DOWN
+    }
+    if not getattr(settings, 'MIT_WS_CERTIFICATE_FILE'):
+        return certificate_status
+
+    try:
+        with open(settings.MIT_WS_CERTIFICATE_FILE, "rt") as certificate:
```
Comment (amir-qayyum-khan) [inline server_status/views.py]: @pdpinch @mitodl/devops is this right settings?

Comment (mbertrand) [reply server_status/views.py]: Might be better to use the `MIT_WS_CERTIFICATE` env variable instead?:
```python
MIT_WS_CERTIFICATE=os.environ.get('MIT_WS_CERTIFICATE', None)
```

and then read in the contents via something like this(based on https://github.com/mitodl/odl-video-service/blob/master/odl_video/envs.py#L155):
```python
certificate = settings.MIT_WS_CERTIFICATE.encode().decode('unicode_escape').encode()
```

**Comment by amir-qayyum-khan** [comment]
Done @mbertrand

File: server_status/views_test.py [hunk 2]
```diff
@@ -61,6 +63,32 @@ def test_view(self):
...
+            SERVICE_UNAVAILABLE
+        )
+    )
+    @unpack
+    @override_settings(HEALTH_CHECK=['CERTIFICATE'])
+    def test_certificate_status_fail(self, certificate_loc, certificate_status, expected_status):
+        """
+        test app certificate expiry status.
+        """
+        with override_settings(MIT_WS_CERTIFICATE=certificate_loc):
```
Comment (mbertrand) [inline server_status/views_test.py]:
```python
with override_settings(MIT_WS_CERTIFICATE=TEST_CERT):
```

File: server_status/views_test.py [hunk 3]
```diff
@@ -61,6 +63,32 @@ def test_view(self):
...
+            assert resp['certificate']["status"] == certificate_status
+
+    def test_status_without_certificate(self):
+        """
+        test app when certificate path in not available in settings.
+        """
+        del settings.MIT_WS_CERTIFICATE
+        resp = self.get(expected_status=SERVICE_UNAVAILABLE)
+        assert 'certificate' not in resp
+
```
Comment (mbertrand) [inline server_status/views_test.py]: Can you also add a `test_certificate_status_pass` test, with `@freeze_time("2019-01-18")` to make sure test passes assuming certificate has expiration date > 30 days after that.

File: server_status/views_test.py [hunk 4]
```diff
@@ -61,6 +63,32 @@ def test_view(self):
...
+        (None, views.DOWN, SERVICE_UNAVAILABLE),
+        (
+            "-----BEGIN CERTIFICATE-----MITX-----END CERTIFICATE-----",
+            views.DOWN,
+            SERVICE_UNAVAILABLE
+        )
+    )
+    @unpack
+    @override_settings(HEALTH_CHECK=['CERTIFICATE'])
+    def test_certificate_status_fail(self, certificate_loc, certificate_status, expected_status):
```
Comment (mbertrand) [inline server_status/views_test.py]: Add `@freeze_time("2020-09-18")` to make sure test fails assuming certificate has expiration date > 30 days before that.

File: test_requirements.txt [hunk 5]
```diff
@@ -1,4 +1,5 @@
 coverage
+ddt
```
Comment (mbertrand) [inline test_requirements.txt]: Add `freezegun` to test with different dates.

**Comment by amir-qayyum-khan** [comment]
Done @mbertrand

**Comment by amir-qayyum-khan** [comment]
Done @mbertrand

**Pull request merged**